service/history/workflow/workflow_task_state_machine.go

```diff
@@ -1124,11 +1124,6 @@ func (m *workflowTaskStateMachine) convertSpeculativeWorkflowTaskToNormal() erro
 	// convert it to normal workflow task before persisting.
 	m.ms.RemoveSpeculativeWorkflowTaskTimeoutTask()
 
-	if !m.ms.IsWorkflowExecutionRunning() {
```

Comment:
is this becase it will block the update state after close?

Reply:
I moved the check to the caller side.